internal/resolution/variable_sources/entity_sources/catalogsource/registry_bundle_to_entity.go

```diff
@@ -0,0 +1,183 @@
+package catalogsource
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/operator-framework/operator-registry/alpha/property"
+	catalogsourceapi "github.com/operator-framework/operator-registry/pkg/api"
+	"k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+)
+
+type UpgradeEdge struct {
+	property.Channel
+	Replaces  string   `json:"replaces,omitempty"`
+	Skips     []string `json:"skips,omitempty"`
+	SkipRange string   `json:"skipRange,omitempty"`
+	Version   string   `json:"version,omitempty"`
+}
+
+type DefaultChannel struct {
+	DefaultChannel string `json:"defaultchannel"`
+}
+
+const (
+	TypeDefaultChannel = "olm.package.defaultChannel"
+	TypeBundleSource   = "olm.bundle.path"
+	TypeLabel          = "olm.label"
+	TypeLabelRequired  = "olm.label.required"
+)
+
+func EntityFromBundle(catsrcID string, pkg *catalogsourceapi.Package, bundle *catalogsourceapi.Bundle) (*input.Entity, error) {
```

Comment:
A good chunk of this code exists in operator-registry here: https://github.com/operator-framework/operator-registry/blob/0aeffa3f44f5e36bd2c0bcc63a94eda000a5f257/pkg/api/api_to_model.go#L12

Is there a way we could re-use this code?